### PR TITLE
MSTR-1142 Scale down operand when there is no active migration

### PR DESCRIPTION
**Pull request by sttts**
To save resources, especially for single-node.

### CODE REVIEW

**Comment by romfreiman** [comment]
/retitle MSTR-1142 Scale down operand when there is no active migration

**Comment by romfreiman** [comment]
Lets add e2e test to verify it?

File: pkg/operator/targetcontroller/sync.go [hunk 1]
```diff
@@ -186,6 +190,27 @@ func (c *TargetController) manageKubeStorageVersionMigratorDeployment(spec *oper
...
+			if (c.Type == migrationv1alpha1.MigrationSucceeded || c.Type == migrationv1alpha1.MigrationFailed) && c.Status == corev1.ConditionTrue {
+				continue
+			}
+
+			foundActive = true
+			break migrationLoop
+		}
+	}
+	if !foundActive {
+		deployment.Spec.Replicas = pointer.Int32Ptr(0)
```
Comment (romfreiman) [inline pkg/operator/targetcontroller/sync.go:211]: can we log on a change?

**Comment by romfreiman** [comment]
@sttts what do you think about reducing the cpu requests even if it's being used - does it really need 100m?

https://github.com/openshift/cluster-kube-storage-version-migrator-operator/search?q=100m

**Comment by marun** [comment]
/retry
/lgtm

Intentional to mix in a vendoring change though?

**Comment by sttts** [comment]
@romfreiman probably we can reduce cpu requests. Feel free to create a PR doing that.

**Comment by romfreiman** [comment]
to what number?

**Comment by sttts** [comment]
Good question. Half of now?

**Comment by sttts** [comment]
On the other hand, do we really care if this operand only starts on demand?

**Comment by romfreiman** [comment]
> On the other hand, do we really care if this operand only starts on demand?

yes, because it will reserve resources and prevent those from user workloads

**Comment by sttts** [comment]
We have to change availability logic. Now it probably expects a pod.

**Pull request closed without merging**